src/MapViewController.swift

```diff
@@ -75,7 +111,30 @@ open class MapViewController: UIViewController, LocationManagerDelegate, TGRecog
       return tgViewController.tilt
     }
   }
-
+  
+  open var panEnabled = true
+  open var pinchEnabled = true
+  open var rotateEnabled = true
+  open var shoveEnabled = true
+  
+  weak open var gestureDelegate: MapGestureDelegate?
+  weak open var panDelegate: MapPanGestureDelegate?
+  weak open var pinchDelegate: MapPinchGestureDelegate?
+  weak open var rotateDelegate: MapRotateGestureDelegate?
+  weak open var shoveDelegate: MapShoveGestureDelegate?
+  
+  weak open var loadDelegate: MapLoadDelegate? {
+    didSet {
+      tgViewController.mapViewDelegate = self
```

Comment:
Nice catch- updated to set delegate to `nil` if both `loadDelegate` and `selectDelegate` are `nil`